### ISSUE TITLE
Ubuntu 14.04 Broken

### ISSUE THREAD

**Issue by e100** [chef-cookbooks/squid#67]
### Cookbook version

2.0.0
### Chef-client version

12.14.77
### Platform Details

Ubuntu 14.04
### Scenario:

Run default recipe
### Steps to Reproduce:

Run default recipe on Ubuntu 14.04
### Expected Result:

Recipe completes without error
### Actual Result:

`  \* template[/etc/squid3/squid.conf] action create (up to date)
- execute[initialize squid cache dir] action run
  # 
  # Error executing action `run` on resource 'execute[initialize squid cache dir]'
  ## Mixlib::ShellOut::ShellCommandFailed
  
  Expected process to exit with [0], but received '1'
  ---- Begin output of squid3 -Nz ----
  STDOUT: 
  STDERR: 2016/09/20 15:21:23| aclParseAclLine: ACL 'manager' already exists with different type.
  FATAL: Bungled /etc/squid3/squid.conf line 7: acl manager proto cache_object
  Squid Cache (Version 3.3.8): Terminated abnormally.
  CPU Usage: 0.004 seconds = 0.004 user + 0.000 sys
  Maximum Resident Size: 373328 KB
  Page faults with physical i/o: 0
  ---- End output of squid3 -Nz ----
  Ran squid3 -Nz returned 1
  ## Cookbook Trace:
  
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'
  ## Resource Declaration:
  # In /var/chef/cache/cookbooks/squid/recipes/default.rb
  
   93: execute 'initialize squid cache dir' do
   94:   command "#{node['squid']['package']} -Nz"
   95:   action :run
   96:   creates ::File.join(node['squid']['cache_dir'], '00')
   97: end
   98: 
  ## Compiled Resource:
  # Declared in /var/chef/cache/cookbooks/squid/recipes/default.rb:93:in `from_file'
  
  execute("initialize squid cache dir") do
    action [:run]
    retries 0
    retry_delay 2
    default_guard_interpreter :execute
    command "squid3 -Nz"
    backup 5
    creates "/var/spool/squid3/00"
    returns 0
    declared_type :execute
    cookbook_name "squid"
    recipe_name "default"
  end
  ## Platform:
  
  x86_64-linux

`

This is the commit that broke it:
https://github.com/chef-cookbooks/squid/commit/b1bf42c4430e1491e8a0e69d5f1de41e0cf201a4

Ubuntu 14.04 uses squid 3.3 not 3.1

**Comment by jerroydmoore** [comment]
I am experiencing this as well.
ChefDK version 0.17.17.
Ubuntu 14.04.

Downgrading/Pinning to 1.1.1 seems to be a good work around until this is resolved.

**Comment by davidkiser** [comment]
Same issue.  Ubuntu 14.04 uses 3.3.8 (currently) by default,  this causes the template to be incompatible

**Comment by bazbremner** [comment]
Note that I raised a PR some time ago to deal with Ubuntu vs Debian some time ago: https://github.com/chef-cookbooks/squid/pull/60

Travis tests failed before I opened that PR and I've not dug further, nor have I had any feedback on the PR.

**Comment by matkam** [comment]
As a workaround until the PR gets merged, I've added the following to the `attributes.rb` file in a wrapper cookbook:
```
if node['platform'] == 'ubuntu'
  if node['platform_version'] == '12.04'
    override['squid']['version'] = '3.1'
  elsif node['platform_version'] == '14.04'
    override['squid']['version'] = '3.3'
  elsif node['platform_version'] == '16.04'
    override['squid']['version'] = '3.5'
  end
end
```

**Comment by cheeseplus** [comment]
Fixed by #50
